programs/protocol-contracts-solana/src/lib.rs

```diff
@@ -56,9 +70,26 @@ pub mod gateway {
         Ok(())
     }
 
+    pub fn update_authority(
+        ctx: Context<UpdateAuthority>,
+        new_authority_address: Pubkey,
+    ) -> Result<()> {
+        let pda = &mut ctx.accounts.pda;
+        require!(
+            ctx.accounts.signer.key() == pda.authority,
+            Errors::SignerIsNotAuthority
+        );
+        pda.authority = new_authority_address;
+        Ok(())
+    }
+
     pub fn deposit(ctx: Context<Deposit>, amount: u64, memo: Vec<u8>) -> Result<()> {
         require!(memo.len() >= 20, Errors::MemoLengthTooShort);
         require!(memo.len() <= 512, Errors::MemoLengthExceeded);
+
+        let pda = &mut ctx.accounts.pda;
```

Comment:
not relating to this PR. Could we also print informative logs in `withdraw`, `deposit_spl_token` and `withdraw_spl_token`, similar to the log print in `withdraw`, which looks so useful (especial to non-technical user).
![image](https://github.com/user-attachments/assets/8708524f-fa54-45ee-8eed-a0fa02e5efc1)